osu.Framework/Graphics/Video/VideoTextureUpload.cs

```diff
@@ -1,40 +1,46 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Framework.Graphics.Textures;
 using osuTK.Graphics.ES30;
 using FFmpeg.AutoGen;
+using osu.Framework.Graphics.Primitives;
+using SixLabors.ImageSharp.PixelFormats;
 
 namespace osu.Framework.Graphics.Video
 {
-    public unsafe class VideoTextureUpload : ArrayPoolTextureUpload
+    public unsafe class VideoTextureUpload : ITextureUpload
```

Comment:
```suggestion
    public sealed unsafe class VideoTextureUpload : ITextureUpload
```
If not implementing the proper disposal pattern with a virtual method.